include/h2o/url.h

```diff
@@ -121,4 +121,27 @@ inline h2o_iovec_t h2o_url_stringify(h2o_mem_pool_t *pool, const h2o_url_t *url)
     return h2o_url_resolve(pool, url, NULL, &tmp);
 }
 
+static inline int h2o_url_host_is_unix_path(h2o_iovec_t host)
+{
+    if (host.len < 6) {
+        return 0;
+    }
+    return h2o_memis(host.base, 6, H2O_STRLIT("unix:/"));
+}
+
+/**
+ * Compares to hostnames, taking into account whether they contain a
+ * unix path (the comparison will be case sensitive) or not.
+ */
+static inline int h2o_url_hosts_are_equal(const h2o_url_t *url_a, const h2o_url_t *url_b)
+{
+    if (url_a->host.len != url_b->host.len)
+        return 0;
+
+    if (h2o_url_host_is_unix_path(url_a->host) && h2o_url_host_is_unix_path(url_b->host))
```

Comment:
I believe calling `h2o_url_host_is_unix_path` only for just one of the two arguments would be sufficient since:
* for the case in which one of the two matches, we will be calling `h2o_memis`, that returns the correct result
* for the case in which one of the two did not match and the other started with `unix:`, then there is no chance for `h2o_lcstris` to return true
* for the case in which one of the two did not match and the other did not start with `unix:`, then calling `h2o_lcstris` is the appropriate action